android/app/src/main/java/com/expensify/chat/customairshipextender/CustomNotificationProvider.java

```diff
@@ -220,30 +210,72 @@ private void applyMessageStyle(@NonNull Context context, NotificationCompat.Buil
                         .setGroupConversation(true)
                         .setConversationTitle(conversationName);
 
+
                 // Add all conversation messages to the notification, including the last one we just received.
-                for (NotificationMessage cachedMessage : notificationData.messages) {
-                    messagingStyle.addMessage(cachedMessage.text, cachedMessage.time, notificationData.getPerson(cachedMessage.accountID));
+                NotificationCompat.MessagingStyle previousStyle = NotificationCompat.MessagingStyle.extractMessagingStyleFromNotification(existingReportNotification.getNotification());
```

Comment:
Yes, at this point, we are certain that a previous notification exists. The `if` statement above checks the `hasExistingNotification` variable, which precisely determines whether the report notification is null or not.

In the case of the first notification, it would bypass this if statement's body and simply add additional data to the extras.

Reply:
You're right, it's the case for the group chats. Thanks for pointing it out